src/controllers/controller.polarArea.js

```diff
@@ -136,9 +137,13 @@ module.exports = DatasetController.extend({
 			start += angle;
 		}
 
-		helpers.each(meta.data, function(arc, index) {
-			me.updateElement(arc, index, reset);
-		});
+		for (i = 0, ilen = arcs.length; i < ilen; ++i) {
+			arcs[i]._options = me._resolveElementOptions(arcs[i], i, reset);
+		}
+
+		for (i = 0, ilen = arcs.length; i < ilen; ++i) {
+			me.updateElement(arcs[i], i, reset);
```

Comment:
Could this be done in the same loop with `_resolveElementOptions`?

Reply:
Yup. Initially I wrote it as two loops to match the other controllers so I could get some tests in. Merging the loops didn't break anything 😄